### PR TITLE
Support session data encryption

### PR DESCRIPTION
**Pull request by chaopeng**
## Related issue

#1649 
@aeneasr 

Currently, Hydra stores claims from consent application as json in database. In our case, we will have a field in /userinfo endpoint stores JWT. Similar to [Aggregated Claims](https://openid.net/specs/openid-connect-core-1_0.html#AggregatedDistributedClaims).

This can be supported by adding claims to id token session in hydra consent application. But hydra will save claims in plain text. Attacker can use the JWT if they grant the database access.

## Proposed changes

1. Add a config flag`db.encrypt_session_data` to hydra
2. If flag is true, encrypt data before writing in database and decrypt after reading from database.
The encryption is using AES via `KeyCipher` in `RegistryBase`.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x] I have read the [security policy](../security/policy)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)
- [x] I have documented my changes in the [developer guide](https://github.com/ory/docs) (if appropriate)

### CODE REVIEW

File: driver/configuration/provider_viper.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ const (
 	ViperKeySubjectTypesSupported          = "oidc.subject_identifiers.supported_types"
 	ViperKeyDefaultClientScope             = "oidc.dynamic_client_registration.default_scope"
 	ViperKeyDSN                            = "dsn"
+	ViperKeyEncryptSessionData             = "db.encrypt_session_data"
```
Comment (aeneasr) [inline driver/configuration/provider_viper.go]: I think we should move this to `oauth2.session.encrypt_at_rest`

**Comment by chaopeng** [comment]
Updated, PTAL, Thank you

**Comment by aeneasr** [comment]
Sorry, traveling right now, will take a look asap

File: oauth2/fosite_store_sql.go [hunk 2]
```diff
@@ -163,9 +174,26 @@ func sqlSchemaFromRequest(signature string, r fosite.Requester, logger logrus.Fi
...
+func maybeJSONObj(s string) bool {
+	if s == "null" {
+		return true
+	}
+	return strings.HasPrefix(s, "{") && strings.HasSuffix(s, "}")
+}
+
+func (s *sqlData) toRequest(session fosite.Session, cm client.Manager, conf Configuration, kc *jwk.AEAD, logger logrus.FieldLogger) (*fosite.Request, error) {
+	sess := s.Session
+	if !maybeJSONObj(string(sess)) {
```
Comment (aeneasr) [inline oauth2/fosite_store_sql.go]: I don't think we're using this library here but gjson is a really good library and has a very fast `gjson.Valid` function that works with all JSON types: https://github.com/tidwall/gjson/blob/master/gjson.go#L2470

```suggestion
	if !gjson.ValidBytes(sess) {
```

Comment (chaopeng) [reply oauth2/fosite_store_sql.go]: This lib looks pretty good. Thank you.

**Comment by aeneasr** [comment]
Awesome 🎉 

Thank you for your contribution!

**Pull request merged**